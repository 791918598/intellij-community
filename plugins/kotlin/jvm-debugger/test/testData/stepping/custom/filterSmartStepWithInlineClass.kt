package filterSmartStepWithInlineClass

@JvmInline
private value class A(val s: String) {
    fun foo(): String {
        return s
    }
}

fun String.bar() = println(this)

fun main() {
    val a = A("a")
    // STEP_INTO: 1
    // STEP_OUT: 1
    // SMART_STEP_INTO_BY_INDEX: 1
    //Breakpoint!
    a.foo().bar()
}
