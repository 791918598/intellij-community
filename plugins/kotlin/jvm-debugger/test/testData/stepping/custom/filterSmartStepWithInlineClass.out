LineBreakpoint created at filterSmartStepWithInlineClass.kt:18
Run Java
Connected to the target VM
filterSmartStepWithInlineClass.kt:18
filterSmartStepWithInlineClass.kt:6
filterSmartStepWithInlineClass.kt:18
filterSmartStepWithInlineClass.kt:10
Disconnected from the target VM

Process finished with exit code 0
